src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java

```diff
@@ -127,39 +127,46 @@ public void onPlayerInteract(PlayerInteractEvent event) {
 				}
 				}
 			}
-		} else if(heldItemStack != null && event.action == Action.RIGHT_CLICK_BLOCK && lookPos != null && isRingEnabled(lokiRing)) {
-			recursion = true;
+		} else if (heldItemStack != null && event.action == Action.RIGHT_CLICK_BLOCK && lookPos != null && isRingEnabled(lokiRing)) {
 			
+			recursion = true;			
+			double oldPosX = player.posX;
+			double oldPosY = player.posY;
+			double oldPosZ = player.posZ;
+
 			for(ChunkCoordinates cursor : cursors) {
 				int x = lookPos.blockX + cursor.posX;
 				int y = lookPos.blockY + cursor.posY;
 				int z = lookPos.blockZ + cursor.posZ;
 				Item item = heldItemStack.getItem();
-				if(!player.worldObj.isAirBlock(x, y, z) && ManaItemHandler.requestManaExact(lokiRing, player, cost, true)) {
-					
+				if (!player.worldObj.isAirBlock(x, y, z) && ManaItemHandler.requestManaExact(lokiRing, player, cost, true)) {
+					player.posX = x;
+					player.posY = y;
+					player.posZ = z;
```

Comment:
This puts the player inside the block that they're accessing, giving them a fixed direction from that block at best and a division by zero at worst. So while this might fix the pure access of a GT block, it might break things like GT tools, Tinkers' hammers (or any other 3x3 tool), or even things that change orientation like Botania's mana spreaders.

Please test these cases at minimum, and consider moving the player to a fixed distance along the viewing axis instead.